dali/operators/generic/reduce/reduce.h

```diff
@@ -144,6 +139,34 @@ class Reduce : public Operator<Backend> {
       false);
     kmgr_.Run<Kernel>(0, 0, ctx, out_view, in_view);
   }
+
+  DALIDataType OutputType() { return output_type_; }
+
+ private:
+  USE_OPERATOR_MEMBERS();
+
+  vector<int> axes_;
+  bool keep_dims_;
+  kernels::KernelManager kmgr_;
+  DALIDataType output_type_ = DALI_NO_TYPE;
+};
+
+
+template <template <typename T, typename R> class ReductionType, typename Backend>
+class ReduceOp : public Reduce<ReductionType, Backend, ReduceOp> {
+ public:
+  explicit inline ReduceOp(const OpSpec &spec) :  Reduce<ReductionType, Backend, ReduceOp>(spec) {}
+
+  void RunImplImpl(workspace_t<Backend> &ws) {
```

Comment:
Because this is not inheritance but CRTP. I can change the name but it is quite accurate. `RunImpl` calls implementations of `RunImplImpl` based on last template parameter.
This way I can share most of the code for reduction and only change `RunImplImpl` based on `ReductionType` template parameter.

Reply:
fair enough